internal/controller/virtualmachine_feedback_controller.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright (c) 2025 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+language governing permissions and limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	clnt "sigs.k8s.io/controller-runtime/pkg/client"
+
+	ckv1alpha1 "github.com/innabox/cloudkit-operator/api/v1alpha1"
+	privatev1 "github.com/innabox/cloudkit-operator/internal/api/private/v1"
+	sharedv1 "github.com/innabox/cloudkit-operator/internal/api/shared/v1"
+)
+
+// VirtualMachineFeedbackReconciler sends updates to the fulfillment service.
+type VirtualMachineFeedbackReconciler struct {
+	logger                  logr.Logger
+	hubClient               clnt.Client
+	virtualMachinesClient   privatev1.VirtualMachinesClient
```

Comment:
This is a normal caching client that only exists because VMs might go on a different cluster than the one this controller is generally watching?